### PR TITLE
React migration: Metadata summary page

### PR DESCRIPTION
**Pull request by antony-britto**
image

### CODE REVIEW

File: app/cdap/components/Metadata/ComplexSchema/index.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+      id: uuidV4(),
+      nested: false,
+      collapse: false,
+    };
+    setParsedSchema([...parsedSchema].splice(placement, 0, newField));
+    formatOutput();
+    if (index !== undefined) {
+      clearTimeout(addFieldTimeout);
+      addFieldTimeout = setTimeout(() => {
+        const elem = document.getElementById(newField.id);
```
Comment (njbriggs) [inline app/cdap/components/Metadata/ComplexSchema/index.tsx]: Can this be a `ref`?

File: app/cdap/components/Metadata/ComplexSchema/index.tsx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      formatOutput();
+    } else {
+      field.collapse = false;
+      field.type = null;
+    }
+
+    field.nested = checkComplexType(field.displayType);
+  }
+
+  function formatOutput(updateDefault = false) {
```
Comment (njbriggs) [inline app/cdap/components/Metadata/ComplexSchema/index.tsx]: Should this logic be pulled out to a reducer? Even using a hook like [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer) might make it cleaner.

**Review by njbriggs — COMMENTED** [review]
Overall, this looks good. There are a couple things you might want to take a look at.

File: app/cdap/components/Metadata/ComplexSchema/index.tsx (hunk 2 above)
Comment (antony-britto) [reply app/cdap/components/Metadata/ComplexSchema/index.tsx]: Done

File: app/cdap/components/Metadata/ComplexSchema/index.tsx (hunk 1 above)
Comment (antony-britto) [reply app/cdap/components/Metadata/ComplexSchema/index.tsx]: Done

File: app/cdap/components/Metadata/ComplexSchema/index.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+const ComplexSchema: React.FC<IComplexSchemaProps> = ({ schema, inputProps }) => {
+  const initialState = {
+    parsedSchema: [],
+    emptySchema: false,
+    error: '',
+    disabledTooltip: null,
+    updateDefault: false,
+    model: {},
+  };
+  function reducer(state, action) {
```
Comment (njbriggs) [inline app/cdap/components/Metadata/ComplexSchema/index.tsx]: At this point, it looks like the reducer isn't doing anything. We should remove the reducer or refactor `getSchemaStore` to be action/reducer based.

Comment (antony-britto) [reply app/cdap/components/Metadata/ComplexSchema/index.tsx]: Since it's a parsing logic moved it a helper method and useState instead of reducer.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**